python/llm/src/bigdl/llm/transformers/model.py

```diff
@@ -248,6 +248,7 @@ def from_pretrained(cls,
             if speculative:
                 from .speculative import speculative_generate, clear_benchmarks
                 # load a sym_int4 model as draft model
+                kwargs["torch_dtype"] = torch.float32
```

Comment:
This change will affect xpu performance. Have fixed it refer to this [pr](https://github.com/intel-analytics/BigDL/pull/10005)

Reply:
why do we need this here？

Reply:
On CPU draft_model need to set `kwargs["torch_dtype"] = torch.float32` to make the query.dtype is the same as key.dtype. Or only can use `(query.to(key.dtype)` to avoid error.

Reply:
> On CPU draft_model need to set `kwargs["torch_dtype"] = torch.float32` to make the query.dtype is the same as key.dtype. Or only can use `(query.to(key.dtype)` to avoid error.

Why does it only affect speculative then?

Reply:
I think the reason is that when only using bigdl-llm int4, user will not set the config `torch_dtype=torch.bfloat16`, but target_model will use it like [this](https://github.com/intel-analytics/BigDL/blob/main/python/llm/example/CPU/Speculative-Decoding/qwen/speculative.py#L65C49-L65C77). And the draft_model is converted from target_model, so it will cause the problem.